Functions - working - Razvan.py

```diff
@@ -0,0 +1,130 @@
+    ('MK', [': ', '58 ', '65 ', '68 ', '69 ', '75 ', '74 ', '79 ', '82 ']),
+    ('MT', ['75 ', '77 ', '78 ', '80 ', '81 ', '81 ', '85 ', '84 ', '86 ']),
+    ('NL', ['94 ', '94 ', '95 ', '96 ', '96 ', '97 ', '98 ', '98 ', '98 ']),
+    ('NO', ['92 ', '93 ', '94 ', '93 ', '97 ', '97 ', '97 ', '96 ', '98 ']),
+    ('PL', ['67 ', '70 ', '72 ', '75 ', '76 ', '80 ', '82 ', '84 ', '87 ']),
+    ('PT', ['58 ', '61 ', '62 ', '65 ', '70 ', '74 ', '77 ', '79 ', '81 ']),
+    ('RO', ['47 ', '54 ', '58 ', '61 b', '68 ', '72 ', '76 ', '81 ', '84 ']),
+    ('RS', [': ', ': ', ': ', ': ', '64 ', ': ', '68 ', '73 ', '80 ']),
+    ('SE', ['91 ', '92 ', '93 ', '90 ', '91 ', '94 b', '95 ', '93 ', '96 ']),
+    ('SI', ['73 ', '74 ', '76 ', '77 ', '78 ', '78 ', '82 ', '87 ', '89 ']),
+    ('SK', ['71 ', '75 ', '78 ', '78 ', '79 ', '81 ', '81 ', '81 ', '82 ']),
+    ('TR', [': ', '47 ', '49 ', '60 ', '70 ', '76 ', '81 ', '84 ', '88 ']),
+    ('UK', ['83 ', '87 ', '88 ', '90 ', '91 ', '93 ', '94 ', '95 ', '96 ']),
+    ('XK', [': ', ': ', ': ', ': ', ': ', ': ', '89 ', '93 ', '93 ']),
+]
+
+type_average_list = ['Year', 'Country']
+
+# create global variables that can be used in perform_average function
+dataset1 = []
+list_year = []
+len_list_year = []
+list_country = []
+len_list_country = []
+
+
+# This function will take the raw dataset as an argument
+# This function will return a dict with the following structure FAIL !!!
+# # {'Romania': [{'year': '2019','coverage': 84}, {'year': '2018','coverage': 67},..., {'year': '2011','coverage': 72}],
+
+def listToDict(rawdata_input):
+    global dataset1
+    # create dataset
+    dataset1 = {country: dict(zip(description[1], coverage)) for country, coverage in rawdata_input}
+    return dataset1
+
+
+# function to retrieve data for each year
+# This function will take the dataset and year as an argument.
+
+def retrieve_data_for_year(year):
+    global len_list_year, dataset1, list_year
+    # loop through dataset values
+    for item1 in dataset1.keys():
+        list_year.append((item1, dataset1[item1][year]))
+        len_list_year = len(list_year)
+    return [f'Data for {year} is: {list_year}']
+
+
+# function to retrieve data for each country
+# This function will take the dataset and country as an argument.
+
+def retrieve_data_for_country(country):
+    global len_list_country, dataset1, list_country
+    for item2 in dataset1[country]:
+        list_country.append((item2, dataset1[country][item2]))
+        len_list_country = len(list_country)
+    return [f'Data for {country} is: {list(list_country)}']
+
+
+# function to perform average from an iterable(of year data or country data)
+# This function will take an iterable as an argument
+# Victor's idea !!!
+def perform_average():
+    global list_year, len_list_year, list_country, len_list_country
+    total = counter = 0
+    for (item, value) in list_year:
+        if value != ': ' and int(value[:-1]) > 0:
```

Comment:
try to filter the values first. By doing this you can use `len()` to gen the final list length and avoid using counter

Reply:
I take all data, strip and transform to integers. Based on this updated list I can make calculations.
I've created 2 functions: average for year and average for country